client/multi-currency-setup/tasks/store-settings-task/index.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External dependencies
+ */
+import React, { useCallback, useContext, useState, useEffect } from 'react';
+import {
+	Button,
+	Card,
+	CardBody,
+	CheckboxControl,
+	Modal,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import interpolateComponents from 'interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import WizardTaskContext from '../../../additional-methods-setup/wizard/task/context';
+import CollapsibleBody from '../../../additional-methods-setup/wizard/collapsible-body';
+import WizardTaskItem from '../../wizard/task-item';
+import './index.scss';
+
+import { useStoreSettings } from 'wcpay/data';
+
+const StoreSettingsTask = () => {
+	const { storeSettings, submitStoreSettingsUpdate } = useStoreSettings();
+
+	const [ status, setStatus ] = useState( 'resolved' );
+
+	const [
+		isAutomaticSwitchEnabledValue,
+		setIsAutomaticSwitchEnabledValue,
+	] = useState( false );
+
+	const [
+		isStorefrontSwitcherEnabledValue,
+		setIsStorefrontSwitcherEnabledValue,
+	] = useState( false );
+
+	const [ isPreviewModalOpen, setPreviewModalOpen ] = useState( false );
+
+	useEffect( () => {
+		if ( Object.keys( storeSettings ).length ) {
+			setIsStorefrontSwitcherEnabledValue(
+				storeSettings.enable_storefront_switcher
+			);
+			setIsAutomaticSwitchEnabledValue(
+				storeSettings.enable_auto_currency
+			);
+		}
+	}, [
+		setIsAutomaticSwitchEnabledValue,
+		setIsStorefrontSwitcherEnabledValue,
+		storeSettings,
+	] );
+
+	const { setCompleted } = useContext( WizardTaskContext );
+
+	const handlePreviewModalOpenClick = useCallback( () => {
+		setPreviewModalOpen( true );
+	}, [ setPreviewModalOpen ] );
+
+	const handlePreviewModalCloseClick = useCallback( () => {
+		setPreviewModalOpen( false );
+	}, [ setPreviewModalOpen ] );
+
+	const handleIsAutomaticSwitchEnabledClick = useCallback(
+		( value ) => {
+			setIsAutomaticSwitchEnabledValue( value );
+		},
+		[ setIsAutomaticSwitchEnabledValue ]
+	);
+
+	const handleIsStorefrontSwitcherEnabledClick = useCallback(
+		( value ) => {
+			setIsStorefrontSwitcherEnabledValue( value );
+		},
+		[ setIsStorefrontSwitcherEnabledValue ]
+	);
+
+	const handleContinueClick = useCallback( () => {
```

Comment:
Same comment as before. Pretty sure you can get rid of those `useCallback` too. We are doing more work memoizing them than allowing the garbage collector to do his work.

Reply:
Whoa! I didn't realized I used that too much :D Thanks for the suggestion.

Reply:
Done in 73cd066.